### PR TITLE
Implement Clock.startTime to get the application's start time

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
Replace a common requirement between old and new logger, using a more appropriate module.

I started to remove old loggers from v3.x.x and it turned out `LogEvent` still had a dependency.
Moving everything to `Clock` seemed like the best compromise.

### CODE REVIEW

File: src/ocean/time/Clock.d [hunk 1]
```diff
@@ -39,6 +39,20 @@ import ocean.core.Exception_tango;
 
 struct Clock
 {
+    /// Time at which the program started
+    private static Time start_time_;
```
Comment (nemanja-boric-sociomantic) [inline src/ocean/time/Clock.d]: Indentation level seems not to match the existing one.

File: src/ocean/util/log/Event.d [hunk 2]
```diff
@@ -40,7 +40,7 @@ public struct LogEvent
     /// Set the various attributes of this event.
     void set (ILogger.Context host, ILogger.Level level, cstring msg, cstring name)
     {
-        time_ = Log.time;
```
Comment (nemanja-boric-sociomantic) [inline src/ocean/util/log/Event.d:43]: Could we deprecate `Log.time` as well?

File: src/ocean/time/Clock.d (hunk 1 above)
Comment (mathias-lang-sociomantic) [reply src/ocean/time/Clock.d]: Ah yes the old Tango modules with their own indentation. Fixed.

File: src/ocean/util/log/Event.d (hunk 2 above)
Comment (mathias-lang-sociomantic) [reply src/ocean/util/log/Event.d:43]: I plan to deprecate the whole `Log` soon.

Comment (mathias-lang-sociomantic) [reply src/ocean/util/log/Event.d:43]: Well actually it cannot hurt, will do

File: src/ocean/time/Clock.d (hunk 1 above)
Comment (nemanja-boric-sociomantic) [reply src/ocean/time/Clock.d]: Thanks!

**Comment by nemanja-boric-sociomantic** [comment]
Also :police_car: is behind you

**Comment by mathias-lang-sociomantic** [comment]
> Also :police_car: is behind you

Why ?

**Comment by nemanja-boric-sociomantic** [comment]
Ha, no, it seems they are there.

**Comment by nemanja-boric-sociomantic** [comment]
Yeah, sorry, I've somehow missed them!

File: src/ocean/util/log/Log.d [hunk 3]
```diff
@@ -334,6 +328,7 @@ public struct Log
 
         ***********************************************************************/
 
+        deprecated("Use ocean.time.Clock : Clock.startTime() directly")
```
Comment (nemanja-boric-sociomantic) [inline src/ocean/util/log/Log.d]: `Clock.now` is the replacement.

**Comment by mihails-strasuns-sociomantic** [comment]
`object.Exception: Cyclic dependency in module ocean.time.WallClock for import ocean.time.Clock` is it related to the PR or just old issue inconvered?

**Comment by mathias-lang-sociomantic** [comment]
> is it related to the PR or just old issue inconvered?

Is "both" a valid answer ? In any case, I fixed it (see first commit, which obviously Github lists last...).

File: src/ocean/util/log/Log.d (hunk 3 above)
Comment (nemanja-boric-sociomantic) [reply src/ocean/util/log/Log.d]: Ping

Comment (mathias-lang-sociomantic) [reply src/ocean/util/log/Log.d]: Oops, fixed

File: relnotes/logevent-started.deprecation.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+* `ocean.util.log.Log : Log.time`
+
+  This function have been deprecated. Use `ocean.time.Clock : Clock.startTime` instead.
```
Comment (nemanja-boric-sociomantic) [inline relnotes/logevent-started.deprecation.md]: Ah, here as well.

Comment (mathias-lang-sociomantic) [reply relnotes/logevent-started.deprecation.md]: At least I was consistent. Fixed

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Comment by mathias-lang-sociomantic** [comment]
Ping

**Comment by nemanja-boric-sociomantic** [comment]
LGTM.

**Pull request merged**